src/vanflow.c

```diff
@@ -484,6 +488,15 @@ static void _vflow_start_record_TH(vflow_work_t *work, bool discard)
 
     vflow_record_t *record = work->record;
 
+    // If this is one of the record types that supports observability,
+    // count it. If the number currently existing is above the threshold,
+    // indicate that we should temporarily stop producing them.
+    if (observability_records[record->record_type]) {
+        if (sys_atomic_inc(&state->observability_record_count) >= 100) {
```

Comment:
This threshold should not be a numeric literal.  At the very least, it should be symbolically defined at the top of the file.  Also, 100 is way too small of a threshold for this feature.  The same applies for the lower limit (20).